### PR TITLE
Improve file layout + API comments

### PR DESCRIPTION
**Pull request by sector-f**

### CODE REVIEW

File: nvc/nvc.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+// EntryFlags describes how a file is stored on disk
+type EntryFlags uint32
+
+const (
+	Uncompressed EntryFlags = 0
+	Compressed   EntryFlags = 1
+	Encrypted    EntryFlags = 3
+
+	magic = "nvc1d\x00\x00\x00"
+)
```
Comment (winny-) [inline nvc/nvc.go:28]: should we document these constants?

File: nvc/nvc.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+	return entries, nil
+
+}
+
+// Data returns from r the file that t is an entry for.
+func (t TocEntry) Data(r io.ReadSeeker) ([]byte, error) {
+	_, err := r.Seek(int64(t.Offset), 0)
+	if err != nil {
+		return []byte{}, err
```
Comment (winny-) [inline nvc/nvc.go:69]: should thiss return `nil, err`?

File: nvc/nvc.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	data := make([]byte, t.RawLength)
+	_, err = io.ReadFull(reader, data)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return data, nil
+}
```
Comment (winny-) [inline nvc/nvc.go:93]: throughout this func it i see returing an empty byte slice on error (`[]byte{}`), should this return `nil` instead?

File: nvc/nvc.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	if err := binary.Read(r, binary.LittleEndian, &count); err != nil {
+		return 0, err
+	}
+	return count, nil
+}
+
+// Hash is a 64-bit FNV-1a hash
+type Hash uint64
+
+func String2Hash(s string) Hash {
```
Comment (winny-) [inline nvc/nvc.go:139]: should this have a doccomment?

File: nvc/nvc.go (hunk 3 above)
Comment (sector-f) [reply nvc/nvc.go:93]: It's not a huge difference, but, yeah, `nil` _is_ preferable. Fixed in b8524d86d5cc79b382e7058dfd4f93b547f645bb

File: nvc/nvc.go (hunk 1 above)
Comment (sector-f) [reply nvc/nvc.go:32]: Yeah, best practice is to document everything that's exported. Done in 8f449c11878837cbb813afc3a0d5b12ee0814b89

Also I renamed the flag constants to be more idiomatic

File: nvc/nvc.go (hunk 4 above)
Comment (sector-f) [reply nvc/nvc.go:144]: Yeah; added in 46f7e0911b0949a48d14d1e07b08c992c06fe1e3

**Review by winny- — APPROVED** [review]
lgtm :boom:

**Pull request merged**